kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterResult.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import org.apache.kafka.common.protocol.ApiMessage;
+
+/**
+ * The result of a filter request or response operation that encapsulates the request or response
+ * to be forwarded to the next filter in the chain.  Optionally it carries orders for actions such
+ * as close the connection or drop the message.
+ *
+ * @see FilterResultBuilder
+ */
+public interface FilterResult {
```

Comment:
On the naming I was wondering if `Command` is a better name than `Result`. The user must decide on a single legal command (forward, drop, forward+close). Then Kroxylicious applies that command. `Result` is often used for "value-or-exception" type holders.